tests/sentry/profiles/test_task.py

```diff
@@ -343,4 +349,4 @@ def test_process_symbolicator_results_for_sample(self):
 
         _process_symbolicator_results_for_sample(profile, stacktraces)
 
-        assert profile["profile"]["stacks"][0] == [0, 1, 2, 3, 4, 5]
+        assert profile["profile"]["stacks"][0] == [5, 0, 1, 4, 3, 2]
```

Comment:
Do I understand this correctly? `0, 1, 2` are the "original" `A, B, C` frames, and we then push new inlined frames in reverse order to the indexed list of frames, and use those indices here?

Is there a specific reason you need to keep the "original" frames in their original order in the global indexed frame list?

Reply:
We are seeing missing indexes when we try to lookup frames (we're trying to read the stack's frame indexes and select the proper frame in the list of frames).

Now, we suspected the client not putting all the frames, we added some checks and validation in `relay` and it didn't seem to be that.

Next was paying attention to this logic of including all inline frames. We blindly replaced our list of frames by the symbolication results and so at first, we realized Symbolicator was giving us more frames than what we sent because of inline frames. We also read in a comment it might remove some frames depending on something (trying to find a link to this comment). So we thought we'd instead carefully merged the results. We're not 100% where the problem is coming from though.

Last, we want the frames in the same order to not have to edit all stacks but just edit the ones with the inline frames.

Reply:
I believe these are two cases where we might skip frames: https://github.com/getsentry/symbolicator/blob/4cac760b8d2315455be6da0304e54129c74f3da5/crates/symbolicator-service/src/services/symbolication/symbolication.rs#L560-L599

Otherwise we should always at least have the frames you put in, even if we fail to symbolicate.